modules/claims_api/app/workers/claims_api/poa_form_builder_job.rb

```diff
@@ -11,12 +11,19 @@ class PoaFormBuilderJob
     include Sidekiq::Worker
     include ClaimsApi::VBMSSidekiq
 
+    # Generate a 21-22 or 21-22a form for a given POA request.
+    # Uploads the generated form to VBMS. If successfully uploaded,
+    # it queues a job to update the POA code in BGS, as well.
+    #
+    # @param power_of_attorney_id [String] Unique identifier of the submitted POA
     def perform(power_of_attorney_id)
       power_of_attorney = ClaimsApi::PowerOfAttorney.find(power_of_attorney_id)
+      poa_code = power_of_attorney.form_data['serviceOrganization']['poaCode']
 
-      output_path = pdf_constructor(power_of_attorney).construct(data(power_of_attorney), id: power_of_attorney.id)
+      output_path = pdf_constructor(poa_code).construct(data(power_of_attorney))
```

Comment:
Do we not need to include that `id:` arg as a 2nd argument to the `construct()` method anymore?
It looks like it'll default to a `SecureRandom.uuid` value if not defined.  But maybe there's value in continuing to allow it to use the id of the POA record as the identifier?  🤷

Reply:
oops, that's a good catch. I think that was an unnecessary removal as I was changing the first call to just have the `poa_code`. I'll revert that in this next commit, too.